api/v1/stats/publications/PKPStatsPublicationHandler.inc.php

```diff
@@ -626,4 +984,194 @@ protected function _processSearchPhrase($searchPhrase, $submissionIds = [])
 
         return $submissionIds;
     }
+
+    /**
+     * Get column names for the submisison CSV report
+     */
+    protected function _getSubmissionReportColumnNames(): array
+    {
+        return [
+            __('common.id'),
+            __('common.title'),
+            __('stats.total'),
+            __('submission.abstractViews'),
+            __('stats.fileViews'),
+            __('stats.pdf'),
+            __('stats.html'),
+            __('common.other'),
+        ];
+    }
+
+    /**
+     * Get column names for the file CSV report
+     */
+    protected function _getFileReportColumnNames(): array
+    {
+        return [
+            __('common.id'),
+            __('common.title'),
+            __('stats.fileViews'),
+            __('common.publication'),
+        ];
+    }
+
+    /**
+     * Get column names for the country, region and city CSV report
+     */
+    protected function _getGeoReportColumnNames(string $scale, bool $withPublication = false): array
+    {
+        $publicationColumns = [];
+        if ($withPublication) {
+            $publicationColumns = [
+                __('common.id'),
+                __('common.title')
+            ];
+        }
+        $scaleColumns = [];
+        if ($scale == StatisticsHelper::STATISTICS_DIMENSION_CITY) {
+            $scaleColumns = [
+                __('stats.city'),
+                __('stats.region')
+            ];
+        } elseif ($scale == StatisticsHelper::STATISTICS_DIMENSION_REGION) {
+            $scaleColumns = [__('stats.region')];
+        } elseif ($scale == StatisticsHelper::STATISTICS_DIMENSION_COUNTRY) {
+            $scaleColumns = [__('common.country'),];
+        }
+        return array_merge(
+            $publicationColumns,
+            $scaleColumns,
+            [__('stats.total'), __('stats.unique')]
+        );
+    }
+
+    /**
+     * Get the CSV row with submission metrics
+     */
+    protected function getCSVItem(int $submissionId, int $abstractViews, int $pdfViews, int $htmlViews, int $otherViews): array
+    {
+        $galleyViews = $pdfViews + $htmlViews + $otherViews;
+        $totalViews = $abstractViews + $galleyViews;
+
+        // Get submission title for display
+        // Now that we use foreign keys, the stats should not exist for deleted submission, but consider it however?
```

Comment:
Ah, this sounds to be complicated :-P
The only solution I can think of is having the soft deletes here too.
But somehow I am not sure if this would be worth... :thinking: 
It is a rare case that someone deletes a submission or something else that was already published -- it is definitely not a good publishing practice...
@asmecher, what do you think?

Reply:
I think I am wrong - to implement the soft deletes for submissions, files, galleys, etc. just because of the statistics is too much.
So if we do need to preserve the statistics for deleted objects the only way is not to use the foreign keys, it seems. Everything else is way too complicated, I believe.
And I am not sure if we do need to keep the stats :thinking:

Reply:
Clinton's opinion is a good idea, but if it helps, I think a step towards better data consistency (foreign keys and `ON CASCADE DELETE`) is a good option. If we need to add soft deletes later, we can do that without promising backwards data availability.

Reply:
OK, I am relieved :-D
I documented the decision in the Github issue, s. https://github.com/pkp/pkp-lib/issues/6782#issuecomment-1136990747.